picky-krb/src/data_types.rs

```diff
@@ -214,6 +216,8 @@ pub struct EncTicketPart {
     pub authorization_data: Optional<Option<ExplicitContextTag10<AuthorizationData>>>,
 }
 
+pub type EncTicketPart = ApplicationTag<EncTicketPartInner, ENC_TICKET_PART_TYPE>;
```

Comment:
issue? The ApplicationTag only relevant in the wrapping structure, so it seems to me that the fix would be to add the ApplicationTag in the wrapping structure directly.

Reply:
> so it seems to me that the fix would be to add the ApplicationTag in the wrapping structure directly.

What do you mean? 😅 

> The ApplicationTag only meaningful in the context of SEQUENCE

yes, and the `EncTicketPartInner` is a SEQUENCE...

Reply:
Context and application tags are only meaningful in the context of another data structure, to distinguish between optional fields. That's why it does not seems appropriate to me to hardcode the tag into the type of the field directly. Maybe in this case it's used at only one place, but the same data structure could be wrapped into a different tag depending on the wrapping structure, so it's appropriate to specify the tag in the enclosing structure.